apps/src/dance/danceRedux.ts

```diff
@@ -108,12 +115,25 @@ export const setSong = createAsyncThunk(
       }
     });
 
-    if (onSongSelected) {
-      onSongSelected(songId);
-    }
+    await handleSongSelection(dispatch, songId, onSongSelected);
   }
 );
 
+async function handleSongSelection(
+  dispatch: ThunkDispatch<unknown, unknown, AnyAction>,
+  songId: string,
+  onSongSelected?: (songId: string) => void
+) {
+  // Temporary branching to support both legacy Dance which manages the current song's
+  // manifest within Dance.js, and Lab2 Dance which reads the current song's manifest from Redux.
+  if (onSongSelected) {
```

Comment:
Good question - in general, I'm hoping we can avoid as much branching as possible. I just added this one to avoid double-loading the song metadata on legacy Dance. I think if we do need an indicator of "is lab2", then checking the lab redux store might suffice? But if there are other cases then yeah it might be good to have a flag somewhere.

Reply:
Cool, we can figure it out down the line.